service/storage_service.cc

```diff
@@ -593,6 +590,8 @@ future<> storage_service::join_token_ring(cdc::generation_service& cdc_gen_servi
         throw std::runtime_error(err);
     }
 
+    assert(_group0);
+    co_await _group0->finish_setup_after_join();
```

Comment:
Is this the only entry point?

Can we listen on the config variable that enables raft and issue SIGHUPs instead of restarts? I want people to have a deluxe experience when upgrading.

Reply:
We could do this as a follow-up. It comes down to removing `SUPPORTS_RAFT_CLUSTER_MANAGEMENT` from disabled features once we toggle the config variable.

Reply:
Great.

Reply:
So it even starts when the cluster agrees it's possible, and if the user forgets to enable it on some node, it won't start and get stuck.

Reply:
Yes, the upgrade procedure starts only if `SUPPORTS_RAFT_CLUSTER_MANAGEMENT` cluster feature is enabled. All `finish_setup_after_join` does is create the feature listener (unless it detects the feature is enabled at this point, then it starts immediately).